index.js

```diff
@@ -59,6 +61,15 @@ async function fastifyStatic (fastify, opts) {
         : prefix + '/'
   }
 
+  if (opts.hash === true) {
+    if (opts.wildcard === undefined || opts.wildcard === true) {
+      throw new Error('"wildcard" has to be disabled to use "hash"')
+    }
+
+    await generateHashForFiles()
```

Comment:
can this potentially cause a timeout when there is a lot of files? maybe there needs to be a flag for making this a background fire-and-forget operation?

Reply:
It could, but I'm not sure how we could make it a background operation, because before this operation is done the server would be unable to serve assets

Reply:
The hashes can now be generated in advance